### PR TITLE
Create a division of install packages with poetry dev vs prod

### LINKED ISSUE #206: Divide required and test dependencies
Currently we can't have a multi stage docker build with just the required dependencies and the test dependencies.
Because the current poetry configuration doesn't have a good separation between the requireed/normal dependencies and the test dependencies.

Probably just split the `[tool.poetry.dependencies]` into itself required and `[tool.poetry.dev-dependencies]`.

### PR DESCRIPTION
**Pull request by jamoqs**
This PR adds poetry option to install prod vs dev using --only main and make default install with this option Closed #206

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -39,6 +34,14 @@ django-celery-results = "2.3.1"
 django-storages = "^1.14.2"
 boto3 = "^1.33.4"
 gunicorn = "21.2.0"
+factory-boy = "^3.3.0"
+pytest-django = "^4.5.2"
```
Comment (igobranco) [inline pyproject.toml:38]: Factory stuff is normal used for testing...
Also the pytest stuff...

**Review by igobranco — CHANGES_REQUESTED** [review]
I think the factory-boy and pytest-django should be on development dependencies.

The Dockerfile should also be changed... in reality this file is the most important to change. Because I won't like to have factory, pytest, black, etc... on the production image.

**Review by Tiago-da-silva23 — APPROVED** [review]

**Comment by igobranco** [comment]
@jamoqs view my comments please and create a new PR.

**Comment by igobranco** [comment]
> @jamoqs view my comments please and create a new PR.

@jamoqs  I'm creating a new PR, because this has broke others that I have been working and were merged concurrently...

**Pull request merged**